backend/inspirehep/curation/views.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+from flask import Blueprint, abort
+from invenio_db import db
+from webargs import fields
+from webargs.flaskparser import FlaskParser
+
+from inspirehep.accounts.decorators import login_required_with_roles
+from inspirehep.accounts.roles import Roles
+from inspirehep.records.api import LiteratureRecord
+from inspirehep.serializers import jsonify
+
+blueprint = Blueprint("inspirehep_curation", __name__, url_prefix="/curation")
+parser = FlaskParser()
+
+
+@parser.error_handler
+def handle_error(error, req, schema, error_status_code, error_headers):
+    message = f"""Incorrect input type for fields: {" ".join(error.field_names)}"""
+    abort(400, message)
+
+
+@blueprint.route("/literature/<int:pid_value>/keywords", methods=["PUT"])
+@login_required_with_roles([Roles.cataloger.value])
+@parser.use_kwargs(
+    {
+        "keywords": fields.List(fields.String, required=False),
+        "_desy_bookkeeping": fields.Dict(required=False),
+    }
+)
+def add_keywords(pid_value, **kwargs):
+    keywords = kwargs.get("keywords", [])
+    desy_bookkeeping = kwargs.get("_desy_bookkeeping", {})
+
+    if not any([keywords, desy_bookkeeping]):
+        return (
+            jsonify(
+                success=False,
+                message="None of required fields was passed",
+            ),
+            400,
+        )
+
+    record = LiteratureRecord.get_record_by_pid_value(pid_value)
+
+    if desy_bookkeeping:
+        record_desy_bookkeeping = record.get("_desy_bookkeeping", [])
+        record_desy_bookkeeping.append(desy_bookkeeping)
```

Comment:
```suggestion
       record.setdefault("_desy_bookkeeping", []).append(desy_bookkeeping)
```
your version has a bug in case `_desy_bookkeeping` is not part of the record yet (you should maybe add a test for that): the first line will create an empty list, the second line will append to it, but that list is never assigned to the record.